src/common/get-file-info.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+const createIgnorer = require("./create-ignorer");
+const options = require("../main/options");
+
+/**
+ * @param {string} filePath
+ * @param {{ ignorePath?: string, withNodeModules?: boolean, plugins: object }} opts
+ *
+ * Please note that prettier.getFileInfo() expects opts.plugins to be an array of paths,
+ * not an object. A transformation from this array to an object is automatically done
+ * internally by the method wrapper. See withPlugins() in index.js.
+ */
+function _getFileInfo(filePath, opts) {
+  let ignored = false;
+  const ignorer = createIgnorer(opts.ignorePath, opts.withNodeModules);
+  ignored = ignorer.ignores(filePath);
+
+  const inferredParser = options.inferParser(filePath, opts.plugins) || null;
+
+  return {
+    ignored,
+    inferredParser
+  };
+}
+
+// the method has been implemented as asynchronous to avoid possible breaking changes in future
+function getFileInfo(filePath, opts) {
+  return new Promise((resolve, reject) => {
+    try {
+      resolve(_getFileInfo(filePath, opts));
+    } catch (e) {
+      reject(e);
+    }
+  });
```

Comment:
This could just be `Promise.resolve().then(() => _getFileInfo(filePath, options))`.

Reply:
I'm happy to include this to https://github.com/prettier/prettier/pull/4451, which has a couple of minor enhancements. See commit log.